setup.py

```diff
@@ -1,114 +1,148 @@
-               "CFBundleIdentifier": "org.cellprofiler.CellProfiler",
-               "CFBundleShortVersionString": cellprofiler.utilities.version.dotted_version,
-               "CFBundleDocumentTypes": [{
-                   "CFBundleTypeExtensions":["cpproj"],
-                   "CFBundleTypeIconFile":"artwork/CellProfilerIcon.icns",
-                   "CFBundleTypeName":"CellProfiler project",
-                   "CFBundleTypeRole":"Editor"
-                   }, {
-                       "CFBundleTypeExtensions":["cppipe"],
-                       "CFBundleTypeIconFile":"artwork/CellProfilerIcon.icns",
-                       "CFBundleTypeName":"CellProfiler pipeline",
-                       "CFBundleTypeRole":"Editor"
-                       }]
-           }
-           }
-
-if sys.argv[-1] == 'py2app':
-    assert not os.path.exists("build"), "Remove the build and dist directories before building app!"
-    assert not os.path.exists("dist"), "Remove the build and dist directories before building app!"
-
-setup(
-    app=APP,
-    package_data={'javabridge':['jars/*.jar']},
-    data_files=DATA_FILES,
-    options={'py2app': OPTIONS},
-    setup_requires=['py2app'],
-    name="CellProfiler"
-)
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        VERSION = "1.0.2"
+
+        directory = os.path.join("imagej", "jars")
+
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        prokaryote = "{}/prokaryote-{}.jar".format(os.path.abspath(directory), VERSION)
+
+        resource = "https://github.com/CellProfiler/prokaryote/" + "releases/download/{tag}/prokaryote-{tag}.jar".format(tag=VERSION)
+
+        request = requests.get(resource, stream=True)
+
+        if not os.path.isfile(prokaryote):
+            with open(prokaryote, "wb") as f:
+                total_length = int(request.headers.get("content-length"))
+
+                for chunk in clint.textui.progress.bar(request.iter_content(chunk_size=1024), expected_size=(total_length/1024) + 1):
+                    if chunk:
+                        f.write(chunk)
+
+                        f.flush()
+
+        dependencies = os.path.abspath("imagej/jars/cellprofiler-java-dependencies-classpath.txt")
 
-if sys.argv[-1] == 'py2app':
-    # there should be some way to do this within setup's framework, but I don't
-    # want to figure it out right now, and our setup is going to be changing
-    # significantly soon, anyway.
-    call('find dist/CellProfiler.app -name tests -type d | xargs rm -rf', shell=True)
-    #call('lipo dist/CellProfiler.app/Contents/MacOS/CellProfiler -thin i386 -output dist/CellProfiler.app/Contents/MacOS/CellProfiler', shell=True)
-    call('rm dist/CellProfiler.app/Contents/Resources/lib/python2.7/cellprofiler/artwork/*.png', shell=True)
+        if not os.path.isfile(dependencies):
+            file = open(dependencies, "w")
+
+            file.write(prokaryote)
+
+            file.close()
+
+
```

Comment:
This is a better idea than cpnose. Need to add code to start the javabridge and stop the javabridge and Ilastik somewhere in here.